### ISSUE TITLE
Add detailed explanation for "Treat Reducers as State Machines"

### ISSUE THREAD

**Issue by bolatovumar** [reduxjs/redux#3692]
## What docs page is being added or updated?

This bit: https://redux.js.org/style-guide/style-guide/#treat-reducers-as-state-machines

- **Section**: #treat-reducers-as-state-machines
- **Page**: Redux Style Guide

### What updates should be made to the page?

Add detailed explanation for "Treat Reducers as State Machines" section

### Do these updates change any of the assumptions or target audience? If so, how do they change?

No, but it would make it easier to understand with a code example provided.

**Comment by bolatovumar** [comment]
I'm willing to submit a PR for this if acked.

**Comment by markerikson** [comment]
Yeah, this ties into #3598 .  I know @davidkpiano had offered to try writing that page.

**Comment by davidkpiano** [comment]
@bolatovumar I can help review, if you'd like! If I were to write it, it would basically be a distillation of this: https://dev.to/davidkpiano/redux-is-half-of-a-pattern-1-2-1hd7 (ignore the article title)

**Comment by bolatovumar** [comment]
@davidkpiano if you are on it then that's great. Your example from the article is great. Your article what got me to open this issue as the docs weren't anywhere as clear as your writeup.
